src/nvim/editor/arbmark.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Same function but names are not limited to one char
+ * There is also no ex_command, just viml functions
+ */
+
+#include "nvim/vim.h"
+#include "nvim/globals.h"
+#include "nvim/mark.h"
+#include "nvim/memory.h"
+#include "nvim/map.h"
+#include "nvim/lib/kvec.h"
+#include "nvim/editor/arbmark.h"
```

Comment:
I would suggest not introducing `editor/` namespace--it is redundant. We have `api/` to separate non-editor functionality, all top-level files are implicitly part of the core editor.

Also consider renaming the file `mark_extended.{h,c}` so that it sorts next to `mark.{h,c}`.